src/components/CompareResults/beta/DownloadButton.tsx

```diff
@@ -13,10 +16,76 @@ const styles = {
   }),
 };
 
+interface GroupedResults {
+  [key: string]: ResultObject[];
+}
+
+interface ResultObject {
+  header_name: string;
+  new_rev: string;
+}
+
 function DownloadButton() {
+  let fileName = 'perf-compare-all-revisions.json';
+  const results = useAppSelector((state) => {
+    if (
+      state.comparison.activeComparison ===
+      Strings.components.comparisonRevisionDropdown.allRevisions
+    ) {
+      return state.compareResults.data;
+    } else {
+      fileName = `perf-compare-${truncateHash(
+        state.comparison.activeComparison,
+      )}.json`;
+      return {
+        [state.comparison.activeComparison]:
+          state.compareResults.data[state.comparison.activeComparison],
+      };
+    }
```

Comment:
This should probably be a memoized selector using `createSelector`, returning the full result of `encodeURIComponent(JSON.stringify(formatDownloadData(results)))`.
Because currently we do all this at each state change, and that doesn't sound right performance-wise, especially if the result data is big.